Library/ViewModels/PostCampaignCheckoutViewModel.swift

```diff
@@ -183,80 +162,41 @@ public class PostCampaignCheckoutViewModel: PostCampaignCheckoutViewModelType,
       .combineLatest(
         checkoutId,
         selectedCard,
-        paymentIntentClientSecretForExistingCards,
-        storedCardsValues
+        paymentIntentClientSecretForExistingCards
       )
 
     // Runs validation for pre-existing cards that were created with setup intents originally but require payment intents for late pledges.
     let validateCheckoutExistingCard = validateCheckoutExistingCardInput
       .takeWhen(self.submitButtonTappedProperty.signal)
-      .filter { _, selectedCard, _, _ in
-        selectedCard.isNewPaymentMethod == false
-      }
-      .switchMap { checkoutId, selectedExistingCreditCard, paymentIntentClientSecret, storedCards in
-        let (_, paymentMethodId, _) = selectedExistingCreditCard
-        let selectedStoredCard = storedCards.first { $0.id == paymentMethodId }
-        let selectedCardStripeCardId = selectedStoredCard?.stripeCardId ?? ""
+      .switchMap { checkoutId, selectedCard, paymentIntentClientSecret in
+
+        assert(
+          selectedCard.stripePaymentMethodId != nil,
+          "Payment method ID should not be missing in a late pledge context."
+        )
 
         return AppEnvironment.current.apiService
           .validateCheckout(
             checkoutId: checkoutId,
-            paymentSourceId: selectedCardStripeCardId,
+            paymentSourceId: selectedCard.stripePaymentMethodId ?? "",
             paymentIntentClientSecret: paymentIntentClientSecret
           )
           .ksr_delay(AppEnvironment.current.apiDelayInterval, on: AppEnvironment.current.scheduler)
           .materialize()
       }
 
-    let validateCheckoutExistingCardSuccess: Signal<PaymentSourceValidation, Never> = Signal
-      .combineLatest(paymentIntentClientSecretForExistingCards, selectedCard, storedCardsValues)
+    self.validateCheckoutSuccess = Signal
+      .combineLatest(paymentIntentClientSecretForExistingCards, selectedCard)
       .takeWhen(validateCheckoutExistingCard.values())
-      .map { paymentIntentClientSecret, selectedCard, storedCards in
-        let (_, paymentMethodId, _) = selectedCard
-        let selectedStoredCard = storedCards.first { $0.id == paymentMethodId }
-        let selectedCardStripeCardId = selectedStoredCard?.stripeCardId ?? ""
+      .map { paymentIntentClientSecret, selectedCard in
 
-        return PaymentSourceValidation(
+        PaymentSourceValidation(
           paymentIntentClientSecret: paymentIntentClientSecret,
-          selectedCardStripeCardId: selectedCardStripeCardId,
+          selectedCardStripeCardId: selectedCard.stripePaymentMethodId,
           requiresConfirmation: true
         )
       }
 
-    // MARK: - Validate New Cards
-
-    let validateCheckoutNewCardInput = Signal.combineLatest(checkoutId, selectedCard)
```

Comment:
All of this can go away, because all cards are now "existing" cards (i.e. stored to the KSR backend)